core/src/smartcontracts/isi/account.rs

```diff
@@ -17,53 +17,61 @@ pub mod isi {
 
     impl<W: WorldTrait> Execute<W> for Mint<Account, PublicKey> {
         type Error = Error;
-        type Diff = DataEvent;
 
         #[metrics(+"mint_account_pubkey")]
         fn execute(
             self,
             _authority: <Account as Identifiable>::Id,
             wsv: &WorldStateView<W>,
-        ) -> Result<Self::Diff, Self::Error> {
-            let public_key = self.object.clone();
+        ) -> Result<Vec<DataEvent>, Self::Error> {
+            let public_key = self.object;
+
             wsv.modify_account(&self.destination_id, |account| {
                 account.signatories.push(public_key);
                 Ok(())
             })?;
-            Ok(self.into())
+
+            Ok(vec![DataEvent::new(
```

Comment:
Also, rather than create a `Vec<DataEvent>` each time we execute an ISI, which slows Iroha down considerably, we could just extend the WSV store with `DataEvents` generated by the WSV modifications inside e.g. `modify_account`.

Reply:
yes, I am aware, but didn't want to make the change as per previous comment

Reply:
The problem with heap allocations is that they're cheap, until they're not.

Reply:
all execution paths were already paying this price. This PR shouldn't bring in any new allocations, only reduce complexity

Reply:
I think that we're agreeing, just not realising it, because it's late. 

I'm 100% on board with reducing complexity.

Reply:
The way I see it, it's either that events are emitted from `Execute::execute` for every ISI or they are returned as is done currently. 

Ok, there is a 3rd possibility, and that is to have `wsv.modify_account` accept the events to be emitted, something like:
```
    pub fn modify_account(
        &self,
        id: &AccountId,
        f: impl FnOnce(&mut Account) -> Result<(), Error>,
        events: Vec<DataEvent>,
    ) -> Result<(), Error>;
```